PR 6549 broke build

PR 6549:
- #6549 

(commit [`29df28d`](https://github.com/assimp/assimp/commit/29df28d7d354dec485fcf2bc4e6a36d44fef0347)) broke build

<details>
<summary>Error details</summary>
<tt>
<...>/assimp/code/AssetLib/USD/USDLoaderImplTinyusdz.cpp:572:14: error: no matching member function for call to 'AddProperty'
  572 |         mat->AddProperty(materialName, AI_MATKEY_NAME);
      |         ~~~~~^~~~~~~~~~~
<...>/assimp/include/assimp/material.h:880:14: note: candidate function not viable: no known conversion from 'aiString' to 'const aiString *' for 1st argument; take the address of the argument with &
  880 |     aiReturn AddProperty(const aiString *pInput,
      |              ^           ~~~~~~~~~~~~~~~~~~~~~~
<...>/assimp/include/assimp/material.inl:237:38: note: candidate function not viable: no known conversion from 'aiString' to 'const float *' for 1st argument
  237 | AI_FORCE_INLINE aiReturn aiMaterial::AddProperty(const float* pInput,
      |                                      ^           ~~~~~~~~~~~~~~~~~~~
<...>/assimp/include/assimp/material.inl:248:38: note: candidate function not viable: no known conversion from 'aiString' to 'const double *' for 1st argument
  248 | AI_FORCE_INLINE aiReturn aiMaterial::AddProperty(const double* pInput,
      |                                      ^           ~~~~~~~~~~~~~~~~~~~~
<...>/assimp/include/assimp/material.inl:259:38: note: candidate function not viable: no known conversion from 'aiString' to 'const aiUVTransform *' for 1st argument
  259 | AI_FORCE_INLINE aiReturn aiMaterial::AddProperty(const aiUVTransform* pInput,
      |                                      ^           ~~~~~~~~~~~~~~~~~~~~~~~~~~~
<...>/assimp/include/assimp/material.inl:270:38: note: candidate function not viable: no known conversion from 'aiString' to 'const aiColor4D *' (aka 'const aiColor4t<float> *') for 1st argument
  270 | AI_FORCE_INLINE aiReturn aiMaterial::AddProperty(const aiColor4D* pInput,
      |                                      ^           ~~~~~~~~~~~~~~~~~~~~~~~
<...>/assimp/include/assimp/material.inl:281:38: note: candidate function not viable: no known conversion from 'aiString' to 'const aiColor3D *' for 1st argument
  281 | AI_FORCE_INLINE aiReturn aiMaterial::AddProperty(const aiColor3D* pInput,
      |                                      ^           ~~~~~~~~~~~~~~~~~~~~~~~
<...>/assimp/include/assimp/material.inl:292:38: note: candidate function not viable: no known conversion from 'aiString' to 'const aiVector3D *' (aka 'const aiVector3t<float> *') for 1st argument
  292 | AI_FORCE_INLINE aiReturn aiMaterial::AddProperty(const aiVector3D* pInput,
      |                                      ^           ~~~~~~~~~~~~~~~~~~~~~~~~
<...>/assimp/include/assimp/material.inl:303:38: note: candidate function not viable: no known conversion from 'aiString' to 'const int *' for 1st argument
  303 | AI_FORCE_INLINE aiReturn aiMaterial::AddProperty(const int* pInput,
      |                                      ^           ~~~~~~~~~~~~~~~~~
<...>/assimp/include/assimp/material.inl:229:22: note: candidate template ignored: could not match 'const TYPE *' against 'aiString'
  229 | aiReturn aiMaterial::AddProperty (const TYPE* pInput,
</tt>
</details>

Part of the problem is that `tinyusdz` isn't built by default, because it fails CI tests.  This means that breakage won't be automatically detected, so developers need to be extra careful when touching the code.

Note: bisected and isolated commit [`29df28d`](https://github.com/assimp/assimp/commit/29df28d7d354dec485fcf2bc4e6a36d44fef0347) as causing the breakage; previous commit [`b10b8ab`](https://github.com/tellypresence/assimp/commit/b10b8ab99a4ccaf3f5af184aa4f26a0dc783b1a0) builds fine.

--- a/code/AssetLib/USD/USDLoaderImplTinyusdz.cpp
+++ b/code/AssetLib/USD/USDLoaderImplTinyusdz.cpp
@@ -511,6 +511,14 @@ void USDImporterImplTinyusdz::uvsForMesh(
     }
 }
 
+static aiColor3D *ownedColorPtrFor(const std::array<float, 3> &color) {
+    aiColor3D *colorPtr = new aiColor3D();
+    colorPtr->r = color[0];
+    colorPtr->g = color[1];
+    colorPtr->b = color[2];
+    return colorPtr;
+}
+
 static std::string nameForTextureWithId(const RenderScene &render_scene, const int targetId) {
     std::stringstream ss;
     std::string texName;
@@ -536,14 +544,14 @@ static void assignTexture(
         const int textureId,
         const int aiTextureType) {
     std::string name = nameForTextureWithId(render_scene, textureId);
-    aiString texName;
-    texName.Set(name);
+    aiString *texName = new aiString();
+    texName->Set(name);
     std::stringstream ss;
     ss.str("");
     ss << "assignTexture(): name: " << name;
     TINYUSDZLOGD(TAG, "%s", ss.str().c_str());
     // TODO: verify hard-coded '0' index is correct
-    mat->AddProperty(&texName, _AI_MATKEY_TEXTURE_BASE, aiTextureType, 0);
+    mat->AddProperty(texName, _AI_MATKEY_TEXTURE_BASE, aiTextureType, 0);
 }
 
 void USDImporterImplTinyusdz::materials(
@@ -567,18 +575,18 @@ void USDImporterImplTinyusdz::materials(
         TINYUSDZLOGD(TAG, "%s", ss.str().c_str());
         aiMaterial *mat = new aiMaterial;
 
-        aiString materialName;
-        materialName.Set(material.name);
+        aiString *materialName = new aiString();
+        materialName->Set(material.name);
         mat->AddProperty(materialName, AI_MATKEY_NAME);
 
         mat->AddProperty(
-                reinterpret_cast<const aiColor3D *>(material.surfaceShader.diffuseColor.value.data()),
+                ownedColorPtrFor(material.surfaceShader.diffuseColor.value),
                 1, AI_MATKEY_COLOR_DIFFUSE);
         mat->AddProperty(
-                reinterpret_cast<const aiColor3D *>(material.surfaceShader.specularColor.value.data()),
+                ownedColorPtrFor(material.surfaceShader.specularColor.value),
                 1, AI_MATKEY_COLOR_SPECULAR);
         mat->AddProperty(
-                reinterpret_cast<const aiColor3D *>(material.surfaceShader.emissiveColor.value.data()),
+                ownedColorPtrFor(material.surfaceShader.emissiveColor.value),
                 1, AI_MATKEY_COLOR_EMISSIVE);
 
         ss.str("");

